esmvaltool/diag_scripts/weighting/calculate_difference_variable_group.py

```diff
@@ -0,0 +1,69 @@
+"""Calculating the difference between two preprocessed files
+ to be further used as other variable groups
+
+"""
+import logging
+import os
+
+from esmvaltool.diag_scripts.shared import (
+    ProvenanceLogger,
+    get_diagnostic_filename,
+    group_metadata,
+    run_diagnostic,
+)
+
+from esmvaltool.diag_scripts.weighting.climwip.io_functions import (
+    read_metadata,
+    read_model_data,
+    read_observation_data,
+    log_provenance,
+)
+
+logger = logging.getLogger(os.path.basename(__file__))
+
+
+def _save_data(data: 'xr.DataArray', name: str, cfg: dict,
+              ancestors: list):
+    """ Save data to netcdf for further use"""
+
+    filename_data = get_diagnostic_filename(
+        '%s%s_ANOM' %(name, data.short_name), cfg, extension='nc')
+    data.to_netcdf(filename_data)
```

Comment:
I think this is the tricky part, right? The code looks good already, but this will save it in the 'work' directory of the 'calculate_local_anomalies' task, not in the preproc folder. Hence, in `climwip/main.py`, it will also not have the same metadata file as is available for the preproc folder. I think the easiest way to deal with this is to add a function to the `io_functions.py` that specifically looks in this location and is able to load this intermediate file in `climwip/main.py`.